modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -480,6 +479,166 @@ inline IE::Blob::Ptr extractBlob(IECallContext& ctx, std::size_t i) {
 }
 } // anonymous namespace
 
+class cv::gimpl::ie::RequestPool {
```

Comment:
I believe this class may become generic as our `concurrent_queue` and other stuff.

Actually, it doesn't even need to care about compiled networks etc.

I'd recommend to move it out to a separate file, refactor it to be some generic "asynchronous task pool", and cover with its own tests. This is the right decomposition to go.

Reply:
Re-designed, but still non generic.